packages/tse-backend/src/modules/trading-strategy/strategies/market-making/market-making.strategy.ts

```diff
@@ -0,0 +1,264 @@
+      clearInterval(strategy.intervalId);
+    }
+
+    this.logger.debug('Paused market making strategy');
+  }
+
+  async stop(command: MarketMakingStrategyActionCommand): Promise<void> {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+
+    await this.marketMakingService.updateStrategyStatusById(
+      strategyEntity.id,
+      StrategyInstanceStatus.STOPPED,
+    );
+
+    const strategy = this.strategies.get(strategyEntity.id);
+    if (strategy) {
+      clearInterval(strategy.intervalId);
+    }
+
+    await this.cancelActiveOrders();
+
+    this.logger.debug(
+      'Stopped market making strategy, not filled orders have been canceled',
+    );
+  }
+
+  async delete(command: MarketMakingStrategyActionCommand) {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+    if (
+      strategyEntity &&
+      strategyEntity.status !== StrategyInstanceStatus.STOPPED
+    ) {
+      throw new BadRequestException(
+        'MarketMaking strategy is not stopped or was deleted',
+      );
+    }
+
+    await this.marketMakingService.updateStrategyStatusById(
+      strategyEntity.id,
+      StrategyInstanceStatus.DELETED,
+    );
+
+    const strategy = this.strategies.get(strategyEntity.id);
+    if (strategy) {
+      clearInterval(strategy.intervalId);
+      this.strategies.delete(strategyEntity.id);
+    }
+
+    this.logger.debug('Soft deleted market making strategy');
+  }
+
+  async start(strategies: MarketMakingStrategyData[]): Promise<void> {
+    this.logger.debug('Starting market making strategy');
+
+    for (const strategy of strategies) {
+      if (!this.strategies.get(strategy.id)) {
+        const intervalId = setInterval(async () => {
+          await this.evaluateMarketMaking(strategy);
+        }, strategy.checkIntervalSeconds * 1000);
+
+        const configuration: StrategyConfig = {
+          strategyKey: createStrategyKey({
+            type: StrategyTypeEnums.MARKET_MAKING,
+            user_id: strategy.userId,
+            client_id: strategy.clientId,
+          }),
+          intervalId,
+          status: StrategyInstanceStatus.RUNNING,
+        };
+        this.strategies.set(strategy.id, configuration);
+      }
+    }
+  }
+
+  async evaluateMarketMaking(command: MarketMakingStrategyCommand) {
+    const {
+      userId,
+      clientId,
+      pair,
+      exchangeName,
+      bidSpread,
+      askSpread,
+      orderAmount,
+      numberOfLayers,
+      priceSourceType,
+      amountChangePerLayer,
+      amountChangeType,
+      ceilingPrice,
+      floorPrice,
+    } = command;
+
+    await this.cancelActiveOrders();
```

Comment:
Why is this here?

Reply:
Cancelling unfilled orders allows the strategy to better adjust new orders to the current market conditions in the next iteration